### PR TITLE
feat(migrations): Enforce presence of Snuba columns in database

### PR DESCRIPTION
**Pull request by lynnagara**
Currently our tests ensure that all database columns must be present in
the corresponding Snuba schema, but not vice versa. This is problematic
since in many cases columns must be added to the database first before they
are referenced in code. A missing database column can cause replacers
or other processes that simply operate on the full list of columns as defined
in the schema to crash.

This flips the check around. The database may have extra columns than the Snuba
schema, but all columns in Snuba's schema must be present in the database.

This came up in https://github.com/getsentry/snuba/pull/1711#issuecomment-784416373

Depends on https://github.com/getsentry/snuba/pull/1718

### CODE REVIEW

**Review by fpacifici — APPROVED** [review]

**Review by fpacifici — CHANGES_REQUESTED** [review]
Sorry, please check the test. I think test_runner will have to be adapted.

**Comment by lynnagara** [comment]
@fpacifici The test should be right, it's failing (correctly) because the extra columns removed in #1718 are still there right now

File: snuba/datasets/schemas/__init__.py [hunk 1]
```diff
@@ -78,22 +78,24 @@ def get_column_differences(
...
                 continue
 
-            expected_type = self.get_columns()[column_name].type
+            expected_type = expected_columns[column.flattened]
 
-            if column.get_raw() != expected_type.get_raw() or column.has_modifier(
+            if column.type.get_raw() != expected_type.get_raw() or column.type.has_modifier(
                 Nullable
-            ) != expected_type.has_modifier(Nullable):
+            ) != expected_type.has_modifier(
```
Comment (evanh) [inline snuba/datasets/schemas/__init__.py:93]: Why does this only check for Nullable? What about the other modifiers?

Comment (lynnagara) [reply snuba/datasets/schemas/__init__.py:93]: Nullable is the only shared modifier between the DB and Snuba schemas. Most of the others e.g. default, codec, low cardinality are only present in the DB version of the schema.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**